Postgres docs (#110)
spiceai/docs: spiceaidocs/content/en/reference/Postgres/_index.md

---
type: docs

> title: "Postgres"

Reviewer: `Postgres` -> `PostreSQL`

linkTitle: "Postgres"
description: 'PostgreSQL reference'
weight: 80
---

PostgreSQL can be connected to Spice AI as either a dataset source, a data store, or a data mesh.

## Dataset Source

> To use PostgreSQL as a dataset source, specify `postgres` as the `source` portion of `from` value for your dataset.

Reviewer: Remove "your" language. Specify as "the" dataset (facts).

> To use PostgreSQL as a dataset source, specify `postgres` as the `source` portion of `from` value for your dataset.

Reviewer 2: `source` is an implementation detail - let's call it the selector:

> To use PostgreSQL as a dataset source, specify `postgres` as the selector in the `from` value for your dataset.

```yaml
datasets:
  - from: postgres:path.to.my_dataset
    name: my_dataset

>     acceleration:

Reviewer: I'd remove the acceleration node here.

        enabled: true
```

> ## Data Mesh

Reviewer: Not sure we need a separate section for data mesh.

> To use PostgreSQL for federated SQL query, specify `postgres` as the `source` portion of `from` value for your dataset, without acceleration.

Reviewer 2: ```suggestion
To use PostgreSQL for federated SQL query, specify `postgres` as the selector in the `from` value for your dataset, without acceleration.
```

```yaml
datasets:
  - from: postgres:path.to.my_dataset
    name: my_dataset
```

## Data Store

> To use PostgreSQL as a data store, specify `postgres` as the `engine` for your dataset.

Reviewer: "as a data store for dataset acceleration"

```yaml
datasets:
  - from: spiceai:path.to.my_dataset
    name: my_dataset
    acceleration:
        engine: postgres
```

## Configuration

> You can configure the connection to your PostgreSQL by providing the following `params`:

Reviewer: Remove "you" and "your". E.g. The PostgreSQL connection is configured with the following parameters.

- `pg_host`: The hostname of the PostgreSQL server.
- `pg_port`: The port of the PostgreSQL server.
- `pg_db`: The name of the database to connect to.
- `pg_user`: The username to connect with.

> - `pg_pass_key`: The name of the secret containing the password to connect with.

Reviewer 2: ```suggestion
- `pg_pass_key`: The name of the secret containing the password to connect with.
```

This is technically the name of the `key` within the `secret` using our nomenclature. The secret is the key-value object.

---
type: docs
title: "PostreSQL"
linkTitle: "PostreSQL"
description: 'PostgreSQL reference'
weight: 80
---

> PostgreSQL can be connected to Spice AI as either a dataset source, a data store, or for federated SQL query.

Reviewer: ```suggestion
PostgreSQL can be used by the Spice runtime as a dataset source, a data store, or for federated SQL query.
```
